crates/arroyo-connectors/src/iggy/sink/mod.rs

```diff
@@ -0,0 +1,160 @@
+        messages: Vec<Message>,
+        ctx: &mut OperatorContext,
+    ) -> Result<()> {
+        if messages.is_empty() {
+            return Ok(());
+        }
+
+        if self.client.is_none() {
+            self.init_client().await?;
+        }
+
+        let client = self.client.as_ref().unwrap();
+        let stream_id = Identifier::from_str(&self.stream)?;
+        let topic_id = Identifier::from_str(&self.topic)?;
+
+        let partitioning = match self.partitioning {
+            PartitioningStrategy::PartitionId(id) => Partitioning::partition_id(id),
+            PartitioningStrategy::Balanced => Partitioning::balanced(),
+        };
+
+        let mut messages_to_send = messages;
+        match client
+            .send_messages(&stream_id, &topic_id, &partitioning, &mut messages_to_send)
+            .await
+        {
+            Ok(_) => {
+                info!("Sent {} messages to Iggy", messages_to_send.len());
+                Ok(())
+            }
+            Err(e) => {
+                error!("Failed to send messages to Iggy: {:?}", e);
+                ctx.report_error("Could not write to Iggy", format!("{:?}", e))
+                    .await;
+
+                sleep(Duration::from_millis(1000)).await;
+                Err(e.into())
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl ArrowOperator for IggySinkFunc {
+    fn name(&self) -> String {
+        format!("iggy-sink-{}-{}", self.stream, self.topic)
+    }
+
+    async fn on_start(&mut self, ctx: &mut OperatorContext) {
+        if let Err(e) = self.init_client().await {
+            error!("Failed to initialize Iggy client: {:?}", e);
+            ctx.report_error("Failed to initialize Iggy client", format!("{:?}", e))
+                .await;
+        }
+
+        if let Some(schema) = ctx.out_schema.as_ref() {
+            info!("Using schema: {:?}", schema);
+        }
+    }
+
+    async fn process_batch(
+        &mut self,
+        batch: RecordBatch,
+        ctx: &mut OperatorContext,
+        _: &mut dyn arroyo_operator::context::Collector,
+    ) {
+        let values = self.serializer.serialize(&batch);
+
+        let timestamps = ctx
+            .out_schema
+            .as_ref()
+            .map(|schema| schema.timestamp_column(&batch));
+
+        let mut messages = Vec::new();
+        for (i, v) in values.into_iter().enumerate() {
+            let _timestamp = timestamps.and_then(|ts| {
+                if ts.is_null(i) {
+                    None
+                } else {
+                    Some(ts.value(i) / 1_000_000)
+                }
+            });
+
+            let message = Message {
+                id: 0,
+                length: v.len() as u32,
+                payload: Bytes::from(v),
+                headers: None,
+            };
+
+            messages.push(message);
+        }
+
+        if let Err(e) = self.send_messages(messages, ctx).await {
+            error!("Failed to send messages to Iggy: {:?}", e);
```

Comment:
This is currently dropping messages on failure, which is generally the wrong default for a stream processing system like arroyo. Instead, we should retry some number of times and then fail the pipeline by panicking.